src/hook-form/useForm.tsx

```diff
@@ -0,0 +1,171 @@
+import { ChangeEvent, useContext, useEffect } from "react";
+import { FormContext } from "./formContext";
+import { ulid } from "ulid";
+import cloneDeep from "lodash.clonedeep";
+
+const splitKeyWithDot = ({ key }: { key: string }) => {
+  return key.split(".");
+};
+
+const getNestedValue = ({ data, targetKeys }) => {
+  const targetKeyArr = splitKeyWithDot({ key: targetKeys });
+
+  return targetKeyArr.reduce((acc, cur) => {
+    // console.log("getNestedValue", { data, acc, cur });
+    if (acc === undefined || cur === undefined) return undefined;
+    if (!isNaN(Number(cur))) return acc[Number(cur)];
+    return acc[cur];
```

Comment:
소소하지만 요렇게 해봐도 좋을 것 같아요! 

```suggestion
    if (obj === undefined) return undefined;
    
    const index = Number(key);
    return !isNaN(index) ? obj[index] : obj[key];
```